### ISSUE TITLE
Enable forking for moj-analytical-services/JDL_RAP

### ISSUE THREAD

**Issue by wcerfgba** [ministryofjustice/data-platform-support#83]
### Which tool do you need help with?

User and Account

### What happened?

Please can you enable forking for the `moj-analytical-services/JDL_RAP` repo? We need to make per-project forks of this repo for making our reports.

**Comment by wcerfgba** [comment]
> Hello, private repository forking is disabled at org level. Would you consider an alternative, for example mirroring?

I'm not sure how to mirror the repo and how this is different from forking, please could you link to some guidance?

The main repo is a template repo and then we need to make changes for each report/analysis that we are commissioned to undertake. However we do like to create PRs back into the main template repo when we make code improvements.

We're also not sure why forking has been disabled, please could you explain why?

**Comment by julialawrence** [comment]
Hiya, you would not be able to fork an internal repo, that is something that's enforced at org level across both `moj-analytical-services` and `ministryofjustice` orgs. You can double-check this in #ask-operations-engineering just to be safe.

Unfortunately, I do not believe you will be natively able to raise PRs on origin repos, if the repos are mirrored. However, an approach might be to add the origin repo as another remote of the target repo and raise PRs that way. It's a bit of additional work (or extra steps as you put it :D) but it would potentially allow you to do what you need. I'd advise testing it out though.

So, I think with forking disallowed, mirroring and using the original repo as another remote on your local machine would get you closest to what you need.

**Comment by wcerfgba** [comment]
Will do, thanks @julialawrence !

**Comment by wcerfgba** [comment]
> Can you make the repository a template and then create new repositories from it? Example[ template repo](https://github.com/ministryofjustice/template-documentation-site) and the [repo](https://github.com/ministryofjustice/technical-guidance)

I do not think this would help as [the Github docs on template repos](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) indicate that:

>  Branches created from a template have unrelated histories, so you cannot create pull requests or merge between the branches.

It seems template repos squash the history so there is a single initial commit for the new repo, and this prevents upstreaming.

**Comment by wcerfgba** [comment]
> @wcerfgba is there a reason this repository cannot be made public immediately?

Currently there is some cleaning up that is required, but also we have separate reports as branches in the template repo, rather than as separate private/internal forked repos. We'd need to be able to move these to forked repos first, tidy up the code and history, including removing the per report repos, and only then could we make the template repo public. Unless we can start to move per-report code out into one or more separate repos, we will never be able to get the repo into a position where we can make it public.

**Comment by wcerfgba** [comment]
Thanks for clarifying @julialawrence , I wasn't sure about how the responsibilities are split across different teams. I'll raise this in the Ops Eng Slack channel and you can close this as deferred. 😄
